### PR TITLE
GTEST: set threshold in tag matching tests

### PR DESCRIPTION
**Pull request by alex-mikheev**
@yosefe please take a look

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4412/ for details (Mellanox internal link).

**Comment by alex-mikheev** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4428/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_tag_match.cc [hunk 1]
```diff
@@ -18,6 +18,7 @@ class test_ucp_tag_match : public test_ucp_tag {
 
     virtual void init()
     {
+        modify_config("TM_THRESH", "1");
```
Comment (shamisp) [inline test/gtest/ucp/test_ucp_tag_match.cc]: what is 1 exactly ?

Comment (alex-mikheev) [reply test/gtest/ucp/test_ucp_tag_match.cc]: We need this to enable hardware tag matching

Comment (shamisp) [reply test/gtest/ucp/test_ucp_tag_match.cc]: 1 - enable , 0 - disable ? 
Is it disabled by default ?

**Comment by miked-mellanox** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4489/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_tag_match.cc (hunk 1 above)
Comment (yosefe) [reply test/gtest/ucp/test_ucp_tag_match.cc]: 1 is the minimal message size for using HW tag-matching

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4571/ for details (Mellanox internal link).

**Pull request merged**